include/abb_librws/rws_poco_client.h

```diff
@@ -304,6 +305,12 @@ namespace rws
      * \brief Container for logging communication results.
      */
     std::deque<HTTPInfo> log_;
+
+    /**
+     * \brief A mutex for preventing concurrent sending requests to controller. This is needed because the controller
+        does not support concurrent requests and sends empty responses or empty headers.
+     */
+    boost::mutex request_mutex_;
```

Comment:
The RWS classes are not expected to be thread-safe. Thread-safety must be ensured on higher level.

Reply:
In general answer I agree.  User also asks for working library and in this case the ABB Controller bug is well hidden and undocumented. I would like to limit user frustration when working with library by implementing thread-safety and informing the user about the reason.

Reply:
> I would like to limit user frustration when working with library by implementing thread-safety and informing the user about the reason.

This approach would require each function and library to be implemented in a thread-safe way, to avoid user frustration.

Reply:
Nah, only those which triggers thread-related problems with controller :)